doc/old/best_practices.rst

```diff
@@ -0,0 +1,121 @@
+
+Introduction
+------------
+
+There are some features of *pydicom* that allow you to help check your code
+for more strict DICOM practices, and to future-proof against major 
+*pydicom* version changes.
+
+It is recommended that you turn on two features if you can: 
+enforcement of valid DICOM, and a flag to enable "future" pydicom changes.
+
+Enforcing Valid DICOM
+---------------------
+
+*pydicom* has a configuration option to help enforce valid DICOM: 
+:attr:`~pydicom.config.enforce_valid_values`.
+
+If set to `True`, some non-standard DICOM will result in *pydicom* raising
+an error rather than assuming a default behavior or issuing a warning.
+
+This flag does not guarantee strict DICOM results, but is a help for 
+some specific situations that have been added to *pydicom* over time.
+
+In some cases, especially if you are dealing with files that do not 
+strictly adhere to the DICOM Standard, you may need to disable this flag.
+
+To turn on the flag in your code:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.enforce_valid_values = True
+
+Note that you *must not* use 
+:code:`from pydicom.config import enforce_valid_values`.
+That makes the `enforce_valid_values` variable local only to that module,
+so *pydicom* would not see your change to its value.
+
+Future-proofing your code
+-------------------------
+
+*pydicom*, like all software, must balance its evolution with not breaking
+existing code using the library. Sometimes, major changes are necessary
+to make significant improvements to the library.
+
+To help you protect your code against future changes, *pydicom* allows you
+to flag that it should behave as it will for any known upcoming 
+major changes.
+
+Running your code with this turned on will help identify any parts of 
+your code that are not compatible with the known changes in the next major
+version of *pydicom*.
+
+The simplest way to set this behavior is to set an environment variable
+``PYDICOM_FUTURE`` to "True". For example to temporarily turn it on in the
+current terminal session:
+
+.. code-block::
+  
+  SET PYDICOM_FUTURE=True           (Windows)
+
+  export PYDICOM_FUTURE=True        (many linux environments)
+
+If you wish to turn off the behavior, you can
+either remove the environment variable, or set it to "False". See your
+operating system documentation for more details on setting or removing
+environment variables.
+
+The other way to enable the future behavior is to turn it on at run-time
+using the :func:`~pydicom.config.future_behavior` function:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.future_behavior()
+
+If you needed to turn the future behavior off again at run-time, you could
+force re-import of the `config` module:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.enforce_valid_values = True
+  # execute code with flag on...
+  # Reload config in its original state
+  import importlib
+  importlib.reload(config)
```

Comment:
I find that a bit cumbersome (e.g. not intuitive) for the user - it would probably nicer to add a flag to `future_behavior` (and maybe rename it to `enable_future_behavior`?), and we could do the reload ourselves.

Reply:
I don't think it is possible (without serious magic) to reload a module from within itself, but perhaps we don't need to - setting all the correct flags should be enough.  However, it means we have to be careful to keep any flags changed in `future_behavior` consistent with their default global values.

Reply:
Updated in latest push.  Docs adjusted.